### PR TITLE
feat: hide features not supported in airgap

### LINKED ISSUE #22006: [Task]: Hide public internet dependent features on air-gapped instances.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by berzerkeer**
## Description

- There are few features which are not supported in Airgapped instance. Such as tempaltes,map widget etc.. 
- This PR adds a check and hides it.

> Hide certain features on airgap instance

Fixes #22006  
> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/editorComponents/GlobalSearch/ResultsNotFound.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { NO_SEARCH_DATA_TEXT } from "@appsmith/constants/messages";
 import { getTypographyByKey } from "design-system-old";
 import { ReactComponent as DiscordIcon } from "assets/icons/help/discord.svg";
 import AnalyticsUtil from "utils/AnalyticsUtil";
+import { isAirgapped } from "ce/utils/airgapHelpers";
```
Comment (ankitakinger) [inline app/client/src/components/editorComponents/GlobalSearch/ResultsNotFound.tsx:8]: import should be `"@appsmith/utils/airgapHelpers"`

Comment (berzerkeer) [reply app/client/src/components/editorComponents/GlobalSearch/ResultsNotFound.tsx:8]: Oh thank you, will update.

**Review by ankitakinger — APPROVED** [review]

**Comment by RakshaKShetty** [comment]
@berzerkeer The snowflake DB is still displayed  which needs to be hidden

**Pull request merged**